src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Orders/Models/Contracts/DeliveryDates/OrderItemRecipientModel.cs

```diff
@@ -9,12 +10,12 @@ public OrderItemRecipientModel()
         {
         }
 
-        public OrderItemRecipientModel(OrderItemRecipient recipient)
+        public OrderItemRecipientModel(OrderRecipient recipient, CatalogueItemId catalogueItemId)
         {
-            CatalogueItemId = recipient.CatalogueItemId;
-            DeliveryDate = recipient.DeliveryDate;
+            CatalogueItemId = catalogueItemId;
+            DeliveryDate = recipient.OrderItemRecipients.FirstOrDefault(x => x.CatalogueItemId == catalogueItemId)?.DeliveryDate;
```

Comment:
Looks like you could call `GetDeliveryDateForItem(...)` instead.
```suggestion
            DeliveryDate = recipient.GetDeliveryDateForItem(catalogueItemId);
```